rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -924,6 +924,83 @@ void saveFederationElection_shouldSaveInStorage() {
+        private final ActivationConfig.ForBlock allActivations = ActivationConfigsForTest.all().forBlock(0L);
+        private final Federation proposedFederation = new P2shErpFederationBuilder().build();
+        private StorageAccessor bridgeStorageAccessor;
+        private FederationStorageProvider federationStorageProvider;
+
+        @BeforeEach
+        void setUp() {
+            bridgeStorageAccessor = new InMemoryStorage();
+            federationStorageProvider = new FederationStorageProviderImpl(bridgeStorageAccessor);
+        }
+
+        @Test
+        void saveProposedFederation_whenProposedFederationIsNotSet_shouldNotSave() {
+            federationStorageProvider.save(networkParameters, allActivations);
+
+            assertNull(getProposedFederationFromRepository());
+            assertNull(getProposedFederationFormatVersionFromRepository());
+        }
+
+        @Test
+        void saveProposedFederation_preRSKIP419_whenProposedFederationIsSet_shouldNotSave() {
+            federationStorageProvider.setProposedFederation(proposedFederation);
+            federationStorageProvider.save(networkParameters, preLovellActivations);
+
+            assertNull(getProposedFederationFromRepository());
+            assertNull(getProposedFederationFormatVersionFromRepository());
+        }
+
+        @Test
+        void saveProposedFederation_whenProposedFederationIsSet_shouldSave() {
+            federationStorageProvider.setProposedFederation(proposedFederation);
+            federationStorageProvider.save(networkParameters, allActivations);
+
+            assertEquals(proposedFederation, getProposedFederationFromRepository());
+            assertEquals(proposedFederation.getFormatVersion(), getProposedFederationFormatVersionFromRepository());
+        }
+
+        @Test
+        void saveProposedFederation_whenProposedFederationIsSetToNull_shouldSave() {
+            // first save a non-null value to make sure saving a null one is actually happening
+            federationStorageProvider.setProposedFederation(proposedFederation);
+            federationStorageProvider.save(networkParameters, allActivations);
+
+            federationStorageProvider.setProposedFederation(null);
+            federationStorageProvider.save(networkParameters, allActivations);
+
+            assertNull(getProposedFederationFromRepository());
+            assertNull(getProposedFederationFormatVersionFromRepository());
+        }
+
+        private Federation getProposedFederationFromRepository() {
+            return bridgeStorageAccessor.getFromRepository(
+                PROPOSED_FEDERATION.getKey(),
+                data -> {
+                    if (data == null) {
+                        return null;
+                    }
+
+                    return BridgeSerializationUtils.deserializeFederationAccordingToVersion(data, getProposedFederationFormatVersionFromRepository(), federationConstants, allActivations);
```

Comment:
How I wish the `BridgeSerializationUtils.deserializeFederationAccordingToVersion` had a `null` check and returns `null` when a `null` is passed so we could get rid of the the null checks like above every time we get form repository :joy: